From: Francois Marier <francois@brave.com>
Date: Tue, 14 Jul 2020 17:39:44 -0700
Subject: Make all favicon requests on-demand

This patch can be dropped once https://crbug.com/1096660 is fixed
as long as the test is retained and still passes.
---
 components/favicon/core/favicon_database.cc | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/components/favicon/core/favicon_database.cc b/components/favicon/core/favicon_database.cc
--- a/components/favicon/core/favicon_database.cc
+++ b/components/favicon/core/favicon_database.cc
@@ -506,6 +506,7 @@ FaviconBitmapID FaviconDatabase::AddFaviconBitmap(
     base::Time time,
     const gfx::Size& pixel_size) {
   DCHECK(icon_id);
+  type = FaviconBitmapType::ON_DEMAND; // Make all icons ON_DEMAND.
 
   sql::Statement statement(db_.GetCachedStatement(
       SQL_FROM_HERE,
@@ -558,8 +559,8 @@ bool FaviconDatabase::SetFaviconBitmap(
   } else {
     statement.BindNull(0);
   }
-  statement.BindInt64(1, time.ToDeltaSinceWindowsEpoch().InMicroseconds());
-  statement.BindInt64(2, 0);
+  statement.BindInt64(1, 0);
+  statement.BindInt64(2, time.ToDeltaSinceWindowsEpoch().InMicroseconds());
   statement.BindInt64(3, bitmap_id);
 
   return statement.Run();
@@ -568,7 +569,7 @@ bool FaviconDatabase::SetFaviconBitmap(
 bool FaviconDatabase::SetFaviconBitmapLastUpdateTime(FaviconBitmapID bitmap_id,
                                                      base::Time time) {
   DCHECK(bitmap_id);
-  // By updating last_updated timestamp, we assume the icon is of type ON_VISIT.
+  /* By updating last_updated timestamp, we assume the icon is of type ON_VISIT.
   // If it is ON_DEMAND, reset last_requested to 0 and thus silently change the
   // type to ON_VISIT.
   sql::Statement statement(
@@ -578,7 +579,8 @@ bool FaviconDatabase::SetFaviconBitmapLastUpdateTime(FaviconBitmapID bitmap_id,
   statement.BindInt64(0, time.ToDeltaSinceWindowsEpoch().InMicroseconds());
   statement.BindInt64(1, 0);
   statement.BindInt64(2, bitmap_id);
-  return statement.Run();
+  return statement.Run(); */
+  return true;
 }
 
 bool FaviconDatabase::SetFaviconsOutOfDateBetween(base::Time begin,
-- 
2.17.1

